### PR TITLE
Adding startup options for bazel

### PR DESCRIPTION
**Pull request by ly0708**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

Adding startup options to bazel

**What is the current behavior? (You can also link to an open issue here)**

There is no options

**What is the new behavior (if this is a feature change)?**

Autocompletion for bazel's options

**Additional info:**

### CODE REVIEW

**Comment by ly0708** [comment]
Hi reviewer. I am not sure why I have 10 commits to merge?

**Comment by fedeci** [comment]
For the next times I'd suggest you not to work on your master branch but always create a new one :)

File: dev/bazel.ts [hunk 1]
```diff
@@ -63,6 +63,209 @@ const completionSpec: Fig.Spec = {
...
   ],
+  options: [
+    {
+      name: "--autodetect_server_javabase",
+      description:
+        "Back to the local JDK for running the bazel server and instead exits",
+    },
+    {
+      name: "--noautodetect_server_javabase",
+      description: "Opposite of --autodetect_server_javabase",
```
Comment (fedeci) [inline dev/bazel.ts:74]: This is `exclusiveOn: ["--autodetect_server_javabase"]`. Can you add a more descriptive description, please?

File: dev/bazel.ts [hunk 2]
```diff
@@ -63,6 +63,209 @@ const completionSpec: Fig.Spec = {
       },
     },
   ],
+  options: [
+    {
+      name: "--autodetect_server_javabase",
+      description:
```
Comment (fedeci) [inline dev/bazel.ts:69]: `exclusiveOn: ["--noautodetect_server_javabase"]`

File: dev/bazel.ts [hunk 3]
```diff
@@ -63,6 +63,209 @@ const completionSpec: Fig.Spec = {
...
+      name: "--noautodetect_server_javabase",
+      description: "Opposite of --autodetect_server_javabase",
+    },
+    {
+      name: "--batch",
+      description: "Run as just a client process without a server",
+    },
+    {
+      name: "--nobatch",
+      description: "Opposite of --batch",
```
Comment (fedeci) [inline dev/bazel.ts:82]: Same as for `--autodetect_server_javabase` about the `exclusiveOn` and I would add more complete description for `--nobatch` too.

ditto, I've seen that the most of the options are like these ones, could you update all of them?

**Review by fedeci — COMMENTED** [review]
A couple of nits but seems ok! Thanks🎉

File: dev/bazel.ts [hunk 4]
```diff
@@ -63,6 +63,209 @@ const completionSpec: Fig.Spec = {
...
+    },
+    {
+      name: "--nobatch_cpu_scheduling",
+      description: "Opposite of --batch_cpu_scheduling",
+    },
+    {
+      name: "--bazelrc",
+      insertValue: "--bazelrc={cursor}",
+      description:
+        "The location of the user .bazelrc file containing default values of Bazel options",
```
Comment (QuiiBz) [inline dev/bazel.ts:96]: Seems like this option takes an argument, and maybe with a template of `filepaths`?

File: dev/bazel.ts [hunk 5]
```diff
@@ -63,6 +63,209 @@ const completionSpec: Fig.Spec = {
...
+    },
+    {
+      name: "--noclient_debug",
+      description: "Opposite of --client_debug",
+    },
+    {
+      name: "--connect_timeout_secs",
+      insertValue: "--connect_timeout_secs={cursor}",
+      description:
+        "The amount of time the client waits for each attempt to connect to the server",
```
Comment (QuiiBz) [inline dev/bazel.ts:120]: Looks like this option also takes an argument!

File: dev/bazel.ts [hunk 6]
```diff
@@ -63,6 +63,209 @@ const completionSpec: Fig.Spec = {
...
+    },
+    {
+      name: "--noexpand_configs_in_place",
+      description: "Opposite of --expand_configs_in_place",
+    },
+    {
+      name: "--failure_detail_out",
+      insertValue: "--failure_detail_out={cursor}",
+      description:
+        "Specifies a location to write a failure_detail protobuf message if the server experiences a failure and cannot report it via gRPC, as normal. Otherwise, the location will be ${OUTPUT_BASE}/failure_detail.rawproto",
```
Comment (QuiiBz) [inline dev/bazel.ts:135]: Same as above

File: dev/bazel.ts [hunk 7]
```diff
@@ -63,6 +63,209 @@ const completionSpec: Fig.Spec = {
...
+    },
+    {
+      name: "--noignore_all_rc_files",
+      description: "Doesn't disables all rc files",
+    },
+    {
+      name: "--io_nice_level={-1,0,1,2,3,4,5,6,7}",
+      insertValue: "--io_nice_level={cursor}",
+      description:
+        "Only on Linux; set a level from 0-7 for best-effort IO scheduling using the sys_ioprio_set system call. 0 is highest priority, 7 is lowest",
```
Comment (QuiiBz) [inline dev/bazel.ts:167]: Same as above. Could you remove the `{-1,0...}` part from the name, and instead add [suggestions](https://fig.io/docs/reference/arg#suggestions) to the argument?

**Review by QuiiBz — CHANGES_REQUESTED** [review]
Thanks for those great additions! Left some comments to resolve before merging this

**Comment by ly0708** [comment]
> For the next times I'd suggest you not to work on your master branch but always create a new one :)

Do you mean always create create a new fork everytime?

**Comment by QuiiBz** [comment]
> Do you mean always create create a new fork everytime?

You just need to create a new branch on your forked repo, and create a PR from this branch

**Comment by ly0708** [comment]
@QuiiBz  You merged this PR, have the changes been made. 
Sorry. I haven't been able to work on this, school has been pretty crazy the past week.

**Comment by QuiiBz** [comment]
> @QuiiBz You merged this PR, but there's no changes yet.

Looks like there were changes: https://github.com/withfig/autocomplete/pull/567/files

> Sorry. I haven't been able to work on this, school has been pretty crazy the past week.
> I see that there's changes to args, so I will work on making the descriptions better.

No worries at all! Feel free to open a new PR when you have the time :)

**Pull request merged**